addons/sale/wizard/sale_generate_payment_link.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class SalePaymentLink(models.TransientModel):
+    _name = "sale.payment.link"
+    _description = "Generate Sales Payment Link"
+
+    @api.model
+    def default_get(self, fields):
+        res = super(SalePaymentLink, self).default_get(fields)
+        active_id = self._context.get('active_id')
+        active_model = self._context.get('active_model')
+        record = self.env[active_model].browse(active_id)
+        invoiced_amount = sum(record.invoice_ids.mapped('amount_total'))
+        res['order_id'] = active_id
+        res['description'] = record.name
+        res['amount'] = record.amount_total - invoiced_amount
+        return res
+
+    order_id = fields.Many2one("sale.order", string="Sales order", require=True)
+    amount = fields.Float(required=True)
+    link = fields.Char(string="Payment link")
+    description = fields.Char(string="Payment Ref")
+
```

Comment:
Use and show currency_id on wizard as monetary widget (or make amount a monetary field)